api/lib/db.js

```diff
@@ -32,6 +32,173 @@ var DB = function(config) {
+
+    function prepareLedgerQuery() {
+      var query = self.knex('ledgers')
+        .select(self.knex.raw("encode(ledgers.ledger_hash, 'hex') as ledger_hash"))
+        .select('ledger_index')
+        .select(self.knex.raw("encode(ledgers.parent_hash, 'hex') as parent_hash"))
+        .select('total_coins')
+        .select('closing_time')
+        .select('close_time_res')
+        .select(self.knex.raw("encode(ledgers.accounts_hash, 'hex') as accounts_hash"))
+        .select(self.knex.raw("encode(ledgers.transactions_hash, 'hex') as transactions_hash"))
+        .orderBy('ledgers.ledger_index', 'desc')
+        .orderBy('closing_time', 'desc')
+        .limit(1);
+      
+      if (!options.ledger_index && !options.datetime && !options.ledger_hash) {
+        query.where('ledgers.closing_time', '<=', moment().unix());
+      }
+      else {
+        if (options.ledger_index) 
+          query.where('ledgers.ledger_index', options.ledger_index);
+        if (options.datetime) {
+          var iso_datetime = moment.utc(options.datetime, moment.ISO_8601);
+          if (iso_datetime.isValid()) {
+            query.where('ledgers.closing_time', '<=', iso_datetime.unix());
+          }
+          else if (!isNaN(options.datetime)) {
+            query.where('ledgers.closing_time', '<=', options.datetime);
+          }
+          else return {error:'invalid datetime, format must be ISO 8601or Unix offset', code:400};
+        }
+        if (options.ledger_hash)
+          query.where('ledgers.ledger_hash', self.knex.raw("decode('"+options.ledger_hash+"', 'hex')"));
+      }
+      return query;
+    }
+
+    function prepareTxQuery(ledger_index) {
+      var query = self.knex('transactions')
+                  .where('transactions.ledger_index', ledger_index);
+
+      if (options.tx_return === 'hex')
+        query.select(self.knex.raw("encode(transactions.tx_hash, 'hex') as tx_hash"));
+      else if (options.tx_return === "binary")
+        query
+          .select(self.knex.raw("encode(transactions.tx_meta, 'hex') as tx_meta"))
+          .select(self.knex.raw("encode(transactions.tx_raw, 'hex') as tx_raw"));
+      else if (options.tx_return === 'json')
+        query.select(self.knex.raw("encode(transactions.tx_raw, 'hex') as tx_raw"))
+          .select(self.knex.raw("encode(transactions.tx_meta, 'hex') as tx_meta"));
+
+      return query;
+    }
+
+    function handleResponse(ledger, transactions) {
+      if (options.tx_return === "hex") {
+        var transaction_list = [];
+        for (var i=0; i<transactions.length; i++){
+          transaction_list.push(transactions[i].tx_hash);
+        }
+        ledger.transactions = transaction_list;
+      }
+      else if (options.tx_return === "binary") ledger.transactions = transactions;
+      else if (options.tx_return === "json") {
+        for (var i=0; i<transactions.length; i++){
+          var row = transactions[i];
+          row.tx = new SerializedObject(row.tx_raw).to_json();
```

Comment:
I think we want to standardize all the transaction responses - there are slight differences for each from rippled's account_tx, ledger, and tx.  This is what I propose:

Transaction
	•	tx
	⁃	raw tx fields
	•	meta
	⁃	metadata fields
	•	hash
	•	date (unix)
	•	ledger_index

Transaction (binary)
	•	tx (binary)
	•	meta (binary)
	•	hash
	•	date (unix)
	•	ledger_index